crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -507,7 +515,9 @@ impl<'a> IRGenerator<'a> {
                 Value::Struct(f)
             }
             (Value::Single(_), Value::Struct(_)) => unreachable!("variables with tuple/struct types should already be decomposed into multiple variables"),
-            (Value::Struct(_), Value::Single(_)) => unreachable!("Uncaught type error, tried to assign a single value to a tuple/struct type"),
+            (Value::Struct(_), Value::Single(_)) => {
+                unimplemented!();
```

Comment:
When you assign to a function returning a structure.

Reply:
In the `(a, b) = foo() ` case, `foo()` is a call instruction so it is a single value.

Reply:
There's the issue then, a call should conceptually return a Value::Struct in that case. In this PR with the results on a separate Results node instead of on the call, assigning `(a, b) = foo()` should have the Results operation as the rhs which should be a struct value.

Reply:
Well no, it _can_ but it can also do otherwise. I suggest we leave this for another PR that will handle multiple return values.

Reply:
The call instruction itself doesn't represent any value in the source program, so the result of `expression_to_object` of a function call expression should be a Value containing any Results from the call rather than the call instruction id itself

Reply:
Since this is not implemented and will be implemented in a different PR about multiple return values, I have restored the previous code.